File: github-miner/src/types/Repository.ts
import { RepositoryOwner } from "./RepositoryOwner";

export type Repository = {
  id?: string;
  name?: string;
  fullPath?: string;
  owner?: RepositoryOwner;
  createdAt?: string;
  updatedAt?: string;
  languages?: string[];
  description?: string;
};

File: GithubAnalyzer/Models/Repository.cs
namespace GithubAnalyzer.Models
{
    public class Repository
    {
        public string Id { get; set; }

        public string Name { get; set; }

        public string FullPath { get; set; }

        public Owner Owner { get; set; }

        public DateTime? CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        public List<Language> Languages { get; set; }

        public string Description { get; set; }
    }
}
